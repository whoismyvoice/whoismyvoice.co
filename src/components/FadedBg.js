import React from 'react';
import cx from 'classnames';

// Component
import BaseComponent from './BaseComponent';

class FadedBG extends BaseComponent {
  constructor() {
    super();
    this._bind('_handleScroll');
    this.state = {
      did_scroll: false
    };
  }

	componentDidMount() {
    window.addEventListener('scroll', this._handleScroll);
  }

  componentWillUnmount() {
  	window.removeEventListener('scroll', this._handleScroll);
  }

  _handleScroll() {
  	if ($(document).scrollTop() > 50) {
  		this._addScrollState();
  	} else {
  		this._disruptScroll();
  	}
  }

  _disruptScroll() {
  	this.setState({
  		did_scroll: false
  	});
  }

  _addScrollState() {
  	this.setState({
  		did_scroll: true
  	});
  }

  render() {
  	const fadedClasses = cx(
      ['faded-bg'],
      {'faded-white': this.props.color === 'white'},
      {'hide': !this.state.did_scroll}
    );
  	return <div className={fadedClasses}></div>;
  }
};

FadedBG.propTypes = {
  color: React.PropTypes.string
}

export default FadedBG;
