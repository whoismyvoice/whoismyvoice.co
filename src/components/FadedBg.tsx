import * as React from 'react';
import * as cx from 'classnames';

interface Props {
  color?: string;
  didScroll?: boolean;
}

class FadedBG extends React.Component<Props> {
  static defaultProps = {
    color: 'white',
  };

  render() {
    const fadedClasses = cx(['faded-bg'], {
      'faded-white': this.props.color === 'white',
      hide: !this.props.didScroll,
    });
    return <div className={fadedClasses} />;
  }
}

export default FadedBG;
