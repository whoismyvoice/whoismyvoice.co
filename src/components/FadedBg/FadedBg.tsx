import React, { VFC } from 'react';
import cx from 'classnames';

interface Props {
  color?: string;
  didScroll?: boolean;
}

export const FadedBg: VFC<Props> = (props) => {
  const { color = 'white', didScroll = false } = props;
  const fadedClasses = cx(['faded-bg'], {
    'faded-white': color === 'white',
    hide: !didScroll,
  });
  return <div className={fadedClasses} />;
};

export default FadedBg;
