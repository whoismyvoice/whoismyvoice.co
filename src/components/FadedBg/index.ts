import Component from './FadedBg';

export default Component;
