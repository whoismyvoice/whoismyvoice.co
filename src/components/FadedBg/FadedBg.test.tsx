import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';
import FadedBg from './FadedBg';

jest.mock('mixpanel-browser');

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<FadedBg />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly no props', () => {
  const tree = renderer.create(<FadedBg />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly color=black', () => {
  const tree = renderer.create(<FadedBg color="black" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly didScroll=false', () => {
  const tree = renderer.create(<FadedBg didScroll={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly didScroll=true', () => {
  const tree = renderer.create(<FadedBg didScroll={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});
