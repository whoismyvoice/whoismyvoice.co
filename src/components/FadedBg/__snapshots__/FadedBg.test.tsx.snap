// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly color=black 1`] = `
<div
  className="faded-bg hide"
/>
`;

exports[`renders correctly didScroll=false 1`] = `
<div
  className="faded-bg faded-white hide"
/>
`;

exports[`renders correctly didScroll=true 1`] = `
<div
  className="faded-bg faded-white"
/>
`;

exports[`renders correctly no props 1`] = `
<div
  className="faded-bg faded-white hide"
/>
`;
